pycbc/events/single.py

```diff
@@ -66,20 +66,33 @@ def insert_args(parser):
                                  'a single value or as detector-value pairs, '
                                  'e.g. H1:mean L1:mean V1:conservative')
 
+    @staticmethod
+    def verify_args(args, parser):
+        sngl_opts = [args.single_fit_file,
+                     args.single_reduced_chisq_threshold,
+                     args.single_duration_threshold,
+                     args.single_newsnr_threshold]
+        sngl_opts_str = ("--single-fit-file, "
+                         "--single-reduced-chisq-threshold, "
+                         "--single-duration-threshold, "
+                         "--single-newsnr-threshold")
+        if args.enable_single_detector_background and not all(sngl_opts):
+            parser.error(f"Single detector trigger options ({sngl_opts_str}) "
+                         "must ALL be given if "
+                         "--enable-single-detector-background is set.")
+        if any(sngl_opts) and not args.enable_single_detector_background:
+            parser.error(f"Single detector trigger options ({sngl_opts_str}) "
+                         "given, but --enable-single-detector-background "
+                         "is not set.")
```

Comment:
This particular check does not look correct, what if the user wants to use `--sngl-ifar-est-dist fixed`? Then `--single-fit-file` is not needed, right?

Reply:
That... is actually a bug right now. So if the fits_file is not given, it will fail even if the `--sngl-ifar-est-dist` is fixed

I'll see if I can fix that

Reply:
This has been fixed, and also changes have been made to allow us to use `--sngl-ifar-est-dist` fixed (and so `--single-fixed-ifar`) for some IFOs, but `mean` or `conservative` for others, which we couldn't do before.

This is then a natural way to switch off single-detector uploads for a specific detector if we wished